rabix-engine/src/main/java/org/rabix/engine/service/impl/JobServiceImpl.java

```diff
@@ -472,13 +478,19 @@ public void handleJobRootAborted(Job rootJob) {
   @Override
   public void handleJobCompleted(Job job){
     logger.info("Job {} rootId: {} is completed.", job.getName(), job.getRootId());
-    if (deleteIntermediaryFiles) {
-      intermediaryFilesService.handleJobCompleted(job);
-    }
     try{
       engineStatusCallback.onJobCompleted(job);
     } catch (EngineStatusCallbackException e) {
       logger.error("Engine status callback failed",e);
     }
+    if (deleteIntermediaryFiles) {
+      Runnable r = new Runnable() {
+        public void run() {
+          intermediaryFilesService.handleJobCompleted(job);
```

Comment:
This shouldn't be done in an asynchronous way. There's a potential problem with deleting some of the data after handling an external event. This needs to be processed from the event processor thread.